src/mjolnir/graphtilebuilder.cc

```diff
@@ -1258,5 +1265,114 @@ void GraphTileBuilder::UpdatePredictedSpeeds(const std::vector<DirectedEdge>& di
+//   }
+
+//   // get the edges edgeinfo, to do that we need to look at the edgeinfo offset of the edge
+//   // so we take the edgeinfo offset from the edge use it as the key into edgeinfo_offset_map_, which
+//   // will give us back an edgeinfobuilder
+//   const auto& edge = directededges_builder_[edge_id.id()];
+//   auto eib = edgeinfo_offset_map_.find(edge.edgeinfo_offset());
+//   // bail here if this iterator is the end
+//   if (eib == edgeinfo_offset_map_.end()) {
+//     throw std::runtime_error("Couldn't find edge info for the given edge: " +
+//                              std::to_string(static_cast<int>(edge_id.id())));
+//   }
+
+//   // we need to use the edgeinfobuilder to add a new tagged value record to it that takes some work
+
+//   // first we construct the string that makes up the record we want to store, ours is going to look
+//   // like:
+//   std::string tagged_value = landmark.to_str();
+
+//   // do we already have this record, if so we dont want a copy instead we just want the offset
+//   auto name_offset = AddName(tagged_value);
+
+//   // so now we know where we are storing this variable sized string in the tile, we need to keep the
+//   // record in the edge info of where that was
+//   NameInfo ni{name_offset, 0, 0, 1};
+//   eib->second->AddNameInfo(ni);
+// }
+
+void GraphTileBuilder::AddLandmark(const GraphId& edge_id, const Landmark& landmark) {
+  // first thing is check that the edge id makes sense, tile id should match and edges_.size() >
+  // edge_id.id if not throw a runtime error. otherwise we grab the edge and then
+  if (header_builder_.graphid().tileid() != edge_id.tileid() ||
+      header_builder_.graphid().level() != edge_id.level()) {
+    throw std::runtime_error("Tile id or hierarchy level doesn't match");
+  }
+  if (header_builder_.directededgecount() <= edge_id.id()) {
+    throw std::runtime_error(
+        "Given edge doesn't exist: edge id is larger than total edge size in this tile");
+  }
+
+  // get the edges edgeinfo, to do that we need to look at the edgeinfo offset of the edge
+  // so we take the edgeinfo offset from the edge use it as the key into edgeinfo_offset_map_, which
+  // will give us back an edgeinfobuilder
+
+  const auto& edge = directededges_builder_[edge_id.id()];
+  const auto original_offset = edge.edgeinfo_offset();
+  auto eib = edgeinfo_offset_map_.find(original_offset);
+  // bail here if this iterator is the end
+  if (eib == edgeinfo_offset_map_.end()) {
+    throw std::runtime_error("Couldn't find edge info for the given edge: " +
+                             std::to_string(static_cast<int>(edge_id.id())));
+  }
+
+  // we need to use the edgeinfobuilder to add a new tagged value record to it that takes some work
+
+  // first we construct the string that makes up the record we want to store, ours is going to look
+  // like:
+  std::string tagged_value = landmark.to_str();
```

Comment:
need a check that this string doesnt include any null chars, throw std::logic_error if so, also need to add this to the AddEdgeInfo function in 3 places above